book/src/components/Ide/index.tsx

```diff
@@ -100,8 +100,9 @@ class DCW {
 
 export type Cursor = { row: number; column: number };
 
+const QUEUE = new Queue();
```

Comment:
Doing it like this means that all `Ide` components will use the same queue. A way to have each component have a different queue that is only created once is to do:
```
const [queue] = useState(() => new Queue());
```

Another option is to move the queue inside `DCW` which is already only created once.

Reply:
Yeah, it makes sense. 

I thought before that `useState` has to return a setter, I also learned how to do lazy initialization from your suggestion. Thanks!

Reply:
Well, `useState` returns an array with 2 elements, and `const [state, setState] = useState(0)` is using [destructuring assignment](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment) to get those values, which is just syntactic sugar for:
```
const _xs = useState(0)
const state = _xs[0]
const setState = _xs[1]
```
so `const [state] = useState(0)` like:
```
const _xs = useState(0)
const state = _xs[0]
```

Reply:
@vemoo You are so sweet! Thanks a lot for the extra explanation 💖